packages/stryker-jasmine-runner/test/unit/JasmineTestRunnerSpec.ts

```diff
@@ -0,0 +1,124 @@
+import * as sinon from 'sinon';
+import { expect } from 'chai';
+import * as helpers from '../../src/helpers';
+import Jasmine = require('jasmine');
+import JasmineTestRunner from '../../src/JasmineTestRunner';
+import { TestResult, TestStatus, RunStatus } from 'stryker-api/test_runner';
+import { expectTestResultsToEqual } from '../helpers/assertions';
+
+type SinonStubbedInstance<TType> = {
+  [P in keyof TType]: TType[P] extends Function ? sinon.SinonStub : TType[P];
+};
+
+describe('JasmineTestRunner', () => {
+
+  let sandbox: sinon.SinonSandbox;
+  let jasmineStub: SinonStubbedInstance<Jasmine>;
+  let evalGlobalStub: sinon.SinonStub;
+  let sut: JasmineTestRunner;
+  let fileNames: ReadonlyArray<string>;
+  let clock: sinon.SinonFakeTimers;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    jasmineStub = sandbox.createStubInstance(Jasmine);
+    jasmineStub.env = {
+      throwOnExpectationFailure: sandbox.stub()
+    };
+    evalGlobalStub = sandbox.stub(helpers, 'evalGlobal');
+    sandbox.stub(helpers, 'Jasmine').returns(jasmineStub);
+    fileNames = Object.freeze(['foo.js', 'bar.js']);
+    clock = sandbox.useFakeTimers();
+    sut = new JasmineTestRunner({ fileNames, strykerOptions: { jasmineConfigFile: 'jasmineConfFile' } });
+  });
+
+  afterEach(() => {
+    delete require.cache['foo.js'];
+    delete require.cache['bar.js'];
+    sandbox.restore();
+  });
+
+  it('should configure jasmine on run', async () => {
+    await actEmptyRun();
+    expect(jasmineStub.execute).called;
+    expect(helpers.Jasmine).calledWithNew;
+    expect(helpers.Jasmine).calledWith({ projectBaseDir: process.cwd() });
+    expect(jasmineStub.loadConfigFile).calledWith('jasmineConfFile');
+    expect(jasmineStub.stopSpecOnExpectationFailure).calledWith(true);
+    expect(jasmineStub.env.throwOnExpectationFailure).calledWith(true);
+    expect(jasmineStub.exit).ok;
+    expect(jasmineStub.clearReporters).called;
+    expect(jasmineStub.randomizeTests).calledWith(false);
+  });
+
+  it('should clear require cache on run', async () => {
+    require.cache['foo.js'] = 'foo';
+    require.cache['bar.js'] = 'bar';
+    await actEmptyRun();
+    expect(require.cache['foo.js']).not.ok;
+    expect(require.cache['bar.js']).not.ok;
+  });
+
+  it('should report completed specs', async () => {
```

Comment:
Could you format the test files so there is an empty line between the `setup`, `act` and `assert` stages to improve readability?

Reply:
👍